src/scenes/Main.cs

```diff
@@ -1,16 +1,34 @@
 using Godot;
 using System;
+using System.Collections.Generic;
 
 public class Main : Node2D
 {
+    private List<Pedestal> pedestals;
+
+    [Signal]
+    delegate void towerSelected(PackedScene towerType);
+    [Signal]
+    delegate void towerDeselected();
 
     public override void _Ready()
     {
         Input.MouseMode = Input.MouseModeEnum.Hidden;
+
+        foreach (Node node in GetChildren())
+        {
+            if (node is Pedestal pedestal) {
+                Connect(nameof(towerSelected), pedestal, "onTowerSelected");
+                Connect(nameof(towerDeselected), pedestal, "onTowerDeselected");
```

Comment:
Consider to use `nameof` for specifying `"onTowerSelected"` and `"onTowerDeselected"` as well.
```suggestion
                Connect(nameof(towerSelected), pedestal, nameof(onTowerSelected));
                Connect(nameof(towerDeselected), pedestal, nameof(onTowerDeselected));
```